RecoEgamma/EgammaTools/python/egammaGainSwitchFixToolsForPAT_cff.py

```diff
@@ -0,0 +1,142 @@
+                      "egmPhotonIDs",
+                      "photonIDValueMapProducer",
+                      "photonRegressionValueMapProducer",
+                      "photonMVAValueMapProducer",
+                      "phPFIsoDepositChargedPAT",
+                      "phPFIsoDepositChargedAllPAT",
+                      "phPFIsoDepositPUPAT",
+                      "phPFIsoDepositNeutralPAT",
+                      "phPFIsoDepositGammaPAT",
+                      "phPFIsoValueCharged04PFIdPAT",
+                      "phPFIsoValueChargedAll04PFIdPAT",
+                      "phPFIsoValuePU04PFIdPAT",
+                      "phPFIsoValueNeutral04PFIdPAT",
+                      "phPFIsoValueGamma04PFIdPAT",
+                      "elPFIsoDepositChargedPAT",
+                      "elPFIsoDepositChargedAllPAT",
+                      "elPFIsoDepositPUPAT",
+                      "elPFIsoDepositNeutralPAT",
+                      "elPFIsoDepositGammaPAT",
+                      "elPFIsoValueCharged04PFIdPAT",
+                      "elPFIsoValueChargedAll04PFIdPAT",
+                      "elPFIsoValuePU04PFIdPAT",
+                      "elPFIsoValueNeutral04PFIdPAT",
+                      "elPFIsoValueGamma04PFIdPAT",
+                      "elPFIsoValueCharged04NoPFIdPAT",
+                      "elPFIsoValueChargedAll04NoPFIdPAT",
+                      "elPFIsoValuePU04NoPFIdPAT",
+                      "elPFIsoValueNeutral04NoPFIdPAT",
+                      "elPFIsoValueGamma04NoPFIdPAT"]
+     
+
+    for name in modules_to_clone:
+        new_name = name+suffix
+        setattr(process,new_name,getattr(process,name).clone())  
+        replace_input_tags(process,new_name,getattr(process,new_name),modules_to_clone,suffix)
+          
+    process.reducedEgammaBeforeGSFix.gsfElectronIDSources = cms.VInputTag(
+        cms.InputTag("eidLoose", processName=cms.InputTag.skipCurrentProcess()),
+        cms.InputTag("eidRobustHighEnergy", processName=cms.InputTag.skipCurrentProcess()),
+        cms.InputTag("eidRobustLoose", processName=cms.InputTag.skipCurrentProcess()),
+        cms.InputTag("eidRobustTight", processName=cms.InputTag.skipCurrentProcess()),
+        cms.InputTag("eidTight", processName=cms.InputTag.skipCurrentProcess()),
+        )
+
+
+
+
+def customizeGSFixForPAT(process): 
+    process.load("RecoEgamma.EgammaTools.egammaGainSwitchFixForPAT_cff")
+    process.load("RecoParticleFlow.PFProducer.pfGSFixLinkerForPAT_cff")
+    process.load("RecoEgamma.EgammaIsolationAlgos.pfClusterIsolationRemapForPAT_cff")
+    process.load("RecoEgamma.ElectronIdentification.idExternalRemapForPAT_cff")
+    process.load("RecoEgamma.EgammaTools.egammaGainSwitchFlag_cff")
+
+    #this clones all the modules before they were modified to run on the orginal collections
+    miniAOD_addOrginalEGamma(process,"BeforeGSFix")
+    process.MINIAODoutput.outputCommands.extend(['keep *_reducedEgammaBeforeGSFix_*_*',
```

Comment:
it's a new method added in #17308  PhysicsTools/PatAlgos/python/tools/helpers.py
so, we get a circular dependency ;)
Better have @gpetruc address it there.

the issue is that the production may assign a different name to the producer. So, you need to try better.
@franzoni do you happen to know if the output module names actually match to what we get with cmsDriver?